packages/gguf/src/gguf.ts

```diff
@@ -232,20 +245,21 @@ export async function gguf(
 
 	const tensorInfos: GGUFTensorInfo[] = [];
 
-	for (let i = 0; i < tensorCount; i++) {
+	for (let i = 0; i < tensorCount.value; i++) {
 		await r.fetchChunkIfNeeded(offset);
 
 		// read tensor name
-		const keyResult = readString(r.view, offset, littleEndian);
+		const keyResult = readString(r.view, offset, version, littleEndian);
 		offset += keyResult.length;
 
 		const nDims = r.view.getUint32(offset, littleEndian);
 		offset += 4;
 
 		const shape: bigint[] = [];
 		for (let dim = 0; dim < nDims; dim++) {
-			shape.push(r.view.getBigUint64(offset, littleEndian));
-			offset += 8;
+			const shape_dim = readVersionedSize(r.view, offset, version, littleEndian);
+			shape.push(shape_dim.value);
+			offset += shape_dim.offset;
```

Comment:
```suggestion
			const shapeDim = readVersionedSize(r.view, offset, version, littleEndian);
			shape.push(shapeDim.value);
			offset += shapeDim.offset;
```
nit: since it is js, the convention is to use camelCase

Reply:
sorry, I've been working with python all day :)

Reply:
same! I writing `print()` in `js`😂

Reply:
with Devin coming we're only doing all of this by hand temporarily anyways 😁

Reply:
Yes. This PR is gonna power devin.gguf